### PR TITLE
Makka / #1801 - set a max allowed decimals on edit listing price

### LINKED ISSUE #1801: [contract v2] contract annoyed by decimals
See here:
Screenshot 2023-10-29 125909

Contract doesn't like too many decimals. Unlikely to ever be a real-world problem, but a bug is a bug I guess...

### PR DESCRIPTION
**Pull request by 0xMakka**
## Description

Ethers throws an error when a user enters more than the maximum allowed decimals on the listing price. This PR limits the max decimals to 6 decimal places.

## Related Ticket

Resolves #1801 

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

File: carbonmark/components/pages/Users/SellerConnected/Forms/EditListing.tsx [hunk 1]
```diff
@@ -155,6 +155,10 @@ export const EditListing: FC<Props> = (props) => {
                     )}`,
                   }),
                 },
+                pattern: {
+                  value: /^\d+(\.\d{1,6})?$/,
+                  message: `The maximum decimals allowed is 6`,
+                },
```
Comment (sprrwhwk) [inline carbonmark/components/pages/Users/SellerConnected/Forms/EditListing.tsx:161]: This should probably accept both comma and dot as decimal separator (both are supported by `<input type="number">`.

See here:
https://codepen.io/selimslim/pen/OJdwrZX

**Review by sprrwhwk — CHANGES_REQUESTED** [review]

File: carbonmark/lib/actions.ts [hunk 2]
```diff
@@ -28,6 +29,10 @@ const getSignerNetwork = (
     : "polygon";
 };
 
+const getFormattedSingleUnitPrice = (value: string) => {
+  return formatToTonnes(value, "en", 6);
+};
```
[the remark this answers is not in the document]
Comment (0xMakka) [reply carbonmark/lib/actions.ts:34]: My bad on this, the name doesn't make sense. At the time I was using the singleUnitPrice and was looking to format it, will rename this function in the morning.

File: carbonmark/components/pages/Users/SellerConnected/Forms/EditListing.tsx (hunk 1 above)
Comment (0xMakka) [reply carbonmark/components/pages/Users/SellerConnected/Forms/EditListing.tsx:161]: Same as above, comma's don't work at all for me in the codepen, even as a replacement for `.`

File: carbonmark/lib/actions.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xMakka) [reply carbonmark/lib/actions.ts]: @0xemc removed the unnecessary `getFormattedSingleUnitPrice` function and used the existing `trimWithLocale` instead.

**Review by 0xemc — APPROVED** [review]

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**